onadata/libs/tests/utils/test_csv_builder.py

```diff
@@ -1209,6 +1210,60 @@ def test_show_choice_labels_select_multiple_2(self, mock_query_data):
         }]
         self.assertEqual(expected_result, result)
 
+    def test_export_data_for_xforms_without_submissions(self):
+        """
+        Test xform schema for form with no submission
+        is successfully exported
+        """
+        fixture = "new_repeats"
+        # publish form so we have a dd
+        self._publish_xls_fixture_set_xform(fixture)
+
+        # Confirm form has not submissions so far
+        self.assertEquals(self.xform.instances.count(), 0)
+        # Generate csv export for form
+        csv_df_builder = CSVDataFrameBuilder(
+            self.user.username, self.xform.id_string, include_images=False)
+        temp_file = NamedTemporaryFile(suffix=".csv", delete=False)
+        csv_df_builder.export_to(temp_file.name)
+        csv_file = open(temp_file.name, 'r')
+        csv_reader = csv.reader(csv_file)
+        header = next(csv_reader)
+
+        expected_header = [
```

Comment:
We are missing some of the repeat headers i.e `kids_name`, `kids_age`... Should this be the case ? Seems okay IMO but curious...

Reply:
Was expecting the headers to be almost identical to:

```
'name', 'age', 'has_kids', 'kids_name', 'kids_age', 'kids_name',
            'kids_age', 'gps', '_gps_latitude', '_gps_longitude',
            '_gps_altitude', '_gps_precision', 'web_browsers/firefox',
            'web_browsers/chrome', 'web_browsers/ie', 'web_browsers/safari',
            'instanceID', '_id', '_uuid', '_submission_time', '_date_modified',
            '_tags', '_notes', '_version', '_duration', '_submitted_by',
            '_total_media', '_media_count', '_media_all_received',
            '_review_status', '_review_comment', '_review_date'
```
but with the group names

Reply:
Yes! I would expect this.
The idea is, if the form contains no submissions, then we only generate the xform headers for export.
I have added an extra test that tests if we generate an export for a form with no submission and then later collect submissions, that the column headers are updated successfully.

Reply:
But in this case, those aren't all the XForm Headers... So should we perhaps add a caveat that repeat headers will not be included in the empty export ?

Reply:
I dont believe repeat headers should be generated within exports for forms with no submissions. Yes.

Reply:
I personally think they should; feels a bit weird to export half the headers because in some scenarios and all in other but I'm fine with whichever approach we decide to take.

Thoughts @ukanga @denniswambua ?

Reply:
The repeat headers tend to be only available when there is data, it is okay if the repeat headers are not included as for CSV exports the repeats are indexed, always.